actionpack/lib/action_dispatch/testing/request_encoder.rb

```diff
@@ -0,0 +1,54 @@
+module ActionDispatch
+  class RequestEncoder # :nodoc:
```

Comment:
Hmm, should we move this to be under `ActionDispatch::TestResponse`? I feel like `ActionDispatch::RequestEncoder` sounds like it's being used in non-test environments. :(